tests/unit/setup_test.go

```
@@ -54,7 +128,20 @@ func CreateTestRouter(store interfaces.LeaderboardStore) *gin.Engine {
 	r.GET("/player/:id/stats", backend.GetPlayerResults(store))
 	r.GET("/leaderboard", backend.GetLeaderboard(store, 10))
 	return r
 }
+
+// Builds a request (with an optional body) and serves it against the router, returning the recorder. body may be "" for GET requests.
+func performRequest(r http.Handler, method, path, body string) *httptest.ResponseRecorder {
+	var reader io.Reader
+	if body != "" {
+		reader = strings.NewReader(body)
+	}
+	req, _ := http.NewRequest(method, path, reader)
```

golangci-lint (2.12.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)